packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx

```diff
@@ -241,8 +241,8 @@ export function emailConfirmationValidation(
 
   return {
     value,
-    valid,
-    error: !valid && !focused ? errorMsg : null,
+    valid: valid || !value,
```

Comment:
emailsMatch in the scope of where this defined is actually a helper method. the boolean emailsMatch that is out of scope but is updated by the function passed in as parameter to this component from the checkout component - thats how it gets updated even though its not directly accessible from here (hopefully that made sense).

Reply:
Hmm I'm not following you here. I am seeing that `emailsMatch` is a boolean state in the parent component, `Checkout`, but it isn't defined anywhere in `NewUserEmailForm`. Its value in the parent component is set when we call `setEmailsMatch`. Using the block-scoped variable `emailsMatch` here won't affect that, but you could change the variable name slightly also if you prefer?

Reply:
Either way, it's not blocking of course.

Reply:
it looks like an unused import from a helpers file at the top of `NewUserEmailForm`:  `import { emailsMatch, isEmailValid } from '../../../../fxa-shared/email/helpers'` which is a function not the boolean variable in the state of the parent component. I'll go ahead and remove the unused import so that `emailsMatch` (the function) isn't in scope anymore and it won't appear with intellisense in vscode.

the boolean `emailsMatch` in the parent component is never passed into `NewUserEmailForm` so its values in inaccessible to be read. The `NewUserEmailForm` takes a setState function which provides a closure around the variable for the child component to update but it'd require a refactor to allow this boolean to be read from the child component - i'm guessing thats why the original author did this `  let valid = false; setEmailsMatch((valid = emailInputState === value));` so that they could work with the same value without having to pass it into the child component.
  
let me know if you'd like me to go forward with that refactor.

Reply:
also let me know if there is another way to access a parent components state without explicitly passing it into the child component. i havent done much work with react and its been a bit since i used vue.